controllers/akodeploymentconfig/cluster/cluster_controller.go

```diff
@@ -54,6 +54,13 @@ type ClusterReconciler struct {
 	GetRemoteClient remote.ClusterClientGetter
 }
 
+// TODO: (adduarte remvoe) AKOO add a finalizer to cluster object ako-operator.networking.tkg.tanzu.vmware.com  which blocks cluster deletion, in case AKO needs to clean up resources in AVI Controller before cluster is deleted.  But some cases, AKO might just crashed or isn't even there, for example, cluster is not created successfully, AKOO will not remove the finalizer since AKO can't report the status correctly.
+// TODO: (adduarte remove) I think first improvement we could do is to check when cluster is deleted, whether the AKO package is installed on the cluster, if not , we could just remove the finalizer regardless.
+func akoInstalled() (bool, error) {
```

Comment:
This should be  changed to a more generic name. 
Something like "checkAKOreadiness" or "checkAKOisnominal", or something else which indicates we are checking weather ako is working as expected.

Reply:
This is a tool function, we'd better move this function to `pkg` folder in case we might want to call it in other controllers in future.